src/cryptonote_core/tx_pool.cpp

```diff
@@ -133,6 +134,15 @@ namespace cryptonote
         return true;
       }
 
+      auto const quorum_type = service_nodes::quorum_type::obligations;
+      auto const quorum_group = service_nodes::quorum_group::worker;
+      crypto::public_key service_node_to_change;
+      if (!service_node_list.try_resolve_pubkey_in_quorum(quorum_type, quorum_group, state_change.block_height, state_change.service_node_index, service_node_to_change))
+      {
+        MERROR("Could not resolve the service node public key from the information in the state change, possibly corrupt tx in your blockchain, rejecting malformed state change");
+        return true;
```

Comment:
It doesn't feel right to return `true` here (or, for that matter, in the case just above it) given the purpose of this function.  It should be perfectly fine for both this (and the above case) to just return false without the MERRORs and just let `check_tx_inputs` (which is also going to get called) deal with the error to reject the transaction.

Reply:
Updated